[src] Introduced x2 new repository rules: git_cached_repository and git_cached_worker

ZachiNachshon opened the pull request.

ZachiNachshon committed:
[src] Introduced x2 new repository rules: git_cached_repository and git_cached_worker

or-shachar commented on git/git_cached_repository.bzl:
```diff
@@ -0,0 +1,222 @@
+        ),
+    "cache_directory":
+        attr.string(
+            default="{}/.git-cache/{}",
```
seems like a template that without values, is it okay?

ZachiNachshon replied:
It is the default value being used to store the git cache in `$HOME/.git-cache/<name-of-the repo>`.
It is string formatted only if no explicit value supplied.

or-shachar commented on git/BUILD.bazel:
```diff
@@ -1 +1,8 @@
\ No newline at end of file
+package(default_visibility = ["//visibility:public"])
+
+filegroup(
```
probably don't need this addition

ZachiNachshon replied:
The `BUILD` file containing a `filegroup` is required as part of the testing framework.

ZachiNachshon replied:
On 2nd though I'll create the `filegroup` within the testing framework dynamically.
Removed the `BUILD` file.

ZachiNachshon replied:
The `BUILD` file is required on the `rules/git` folder when `git_cached_repository` is loaded before usage. 
Reverting change and adjusting tests accordingly.

```
ERROR: error loading package '': 
Every .bzl file must have a corresponding package, but '@wix_build_tools//rules/git:git_cached_repository.bzl' does not have one.
```

or-shachar commented on git/git_cached_repository.bzl:
```diff
@@ -0,0 +1,222 @@
+        Returns:
+            Path to the local repository cache folder
+    """
+    if "{}/.git-cache/{}" in repo_ctx.attr.cache_directory:
```
it feels kinda awkward that this is the default value, maybe it's better to actually set default value to `None` and then this line would look like:
```suggestion
    if not repo_ctx.attr.cache_directory:
```

ZachiNachshon replied:
Set `cache_directory ` default to `None` and using `{}/.git-cache/{}` path explicitly if no path supplied.

or-shachar reviewed (changes requested):
overall as initial version , seems good.
I'd run `buildifier` on it to make sure it's well linted.

Please see my comments, and then you can merge

or-shachar commented:
I integrated this in my repo using:
```python
wix_build_tools_version="eef838fb0ba68f33534a1bc3dc1b33a1c30cfd29"
http_archive(
    name = "wix_build_tools",
    urls = ["https://github.com/wix-playground/wix_build_tools/archive/%s.zip" % wix_build_tools_version],
    strip_prefix = "wix_build_tools-%s" % wix_build_tools_version,
    sha256 = "1383685501aac716c9abda4f83bacf43b331d5cd38b5e2d3933c5f57afc29b48",
)
```

And then to use `git_cached_repository` I used:
```python
load("@wix_build_tools//git:git_cached_repository.bzl", "git_cached_repository")

```

ZachiNachshon committed:
[src] git cached repo - removed cache_directory CLI args default value, if none supplied use a pre-defined cache path
[src] Removed redundant filegroup from git rules folder
[src] Buildifier on both git rules
